nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java

```diff
@@ -112,13 +114,28 @@ public Description createErrorDescription(
     return builder.build();
   }
 
-  private boolean isOptionalTypeErrorAndSuppressed(
-      ErrorMessage errorMessage, @Nullable Tree suggestTree) {
-    return errorMessage.messageType.equals(GET_ON_EMPTY_OPTIONAL)
-        && suggestTree != null
-        && ASTHelpers.getSymbol(suggestTree) != null
-        && symbolHasSuppressWarningsAnnotation(
-            ASTHelpers.getSymbol(suggestTree), OPTIONAL_CHECK_NAME);
+  /**
+   * Find out if a particular subchecker (e.g. NullAway.Optional) is being suppressed in a given
+   * path.
+   *
+   * <p>This requires a tree path traversal, which is expensive, but we only do this when we would
+   * otherwise report an error, which means this won't happen for most nodes/files.
+   *
+   * @param treePath The path with the error location as the leaf.
+   * @param subcheckerName The string to check for inside @SuppressWarnings
+   * @return
+   */
+  private boolean hasPathSuppression(TreePath treePath, String subcheckerName) {
+    return StreamSupport.stream(treePath.spliterator(), false)
+        .filter(
```

Comment:
Can we extract this filter predicate (which is also used in `suppressibleNode()`) into a named function and call it from both places?

Reply:
Done!